mix compiler error

Just added vintage_net to (`~> 0.2.4`) an existing project and was greeted with this compiler error:
```
==> vintage_net
mkdir -p /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj
mkdir -p /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../priv
/home/connor/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-linux_x86_64-1.1.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/erts-10.3.4/include -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/lib/erl_interface-3.11.2/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include -std=gnu99 -o /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj/to_elixir.o src/to_elixir.c
/home/connor/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-linux_x86_64-1.1.0/bin/arm-unknown-linux-gnueabihf-gcc /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj/to_elixir.o -L/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/lib/erl_interface-3.11.2/lib -lei_st --sysroot=/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging -o /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../priv/to_elixir
/home/connor/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-linux_x86_64-1.1.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/erts-10.3.4/include -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/lib/erl_interface-3.11.2/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include -std=gnu99 -o /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj/udhcpc_handler.o src/udhcpc_handler.c
/home/connor/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-linux_x86_64-1.1.0/bin/arm-unknown-linux-gnueabihf-gcc /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj/udhcpc_handler.o -L/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/lib/erl_interface-3.11.2/lib -lei_st --sysroot=/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging -o /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../priv/udhcpc_handler
/home/connor/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-linux_x86_64-1.1.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/erts-10.3.4/include -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/lib/erlang/lib/erl_interface-3.11.2/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include -std=gnu99 -o /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj/if_monitor.o src/if_monitor.c
In file included from src/if_monitor.c:30:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:26:0: warning: "IFNAMSIZ" redefined
 #define IFNAMSIZ 16
 
In file included from src/if_monitor.c:28:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:129:0: note: this is the location of the previous definition
 # define IFNAMSIZ IF_NAMESIZE
 
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:45:17: error: redeclaration of enumerator 'IFF_UP'
 # define IFF_UP IFF_UP
                 ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:71:2: note: in expansion of macro 'IFF_UP'
  IFF_UP    = 1<<0,  /* sysfs */
  ^~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:44:5: note: previous definition of 'IFF_UP' was here
     IFF_UP = 0x1,  /* Interface is up.  */
     ^~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:47:24: error: redeclaration of enumerator 'IFF_BROADCAST'
 # define IFF_BROADCAST IFF_BROADCAST
                        ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:72:2: note: in expansion of macro 'IFF_BROADCAST'
  IFF_BROADCAST   = 1<<1,  /* __volatile__ */
  ^~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:46:5: note: previous definition of 'IFF_BROADCAST' was here
     IFF_BROADCAST = 0x2, /* Broadcast address valid.  */
     ^~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:49:20: error: redeclaration of enumerator 'IFF_DEBUG'
 # define IFF_DEBUG IFF_DEBUG
                    ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:73:2: note: in expansion of macro 'IFF_DEBUG'
  IFF_DEBUG   = 1<<2,  /* sysfs */
  ^~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:48:5: note: previous definition of 'IFF_DEBUG' was here
     IFF_DEBUG = 0x4,  /* Turn on debugging.  */
     ^~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:51:23: error: redeclaration of enumerator 'IFF_LOOPBACK'
 # define IFF_LOOPBACK IFF_LOOPBACK
                       ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:74:2: note: in expansion of macro 'IFF_LOOPBACK'
  IFF_LOOPBACK   = 1<<3,  /* __volatile__ */
  ^~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:50:5: note: previous definition of 'IFF_LOOPBACK' was here
     IFF_LOOPBACK = 0x8,  /* Is a loopback net.  */
     ^~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:53:26: error: redeclaration of enumerator 'IFF_POINTOPOINT'
 # define IFF_POINTOPOINT IFF_POINTOPOINT
                          ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:75:2: note: in expansion of macro 'IFF_POINTOPOINT'
  IFF_POINTOPOINT   = 1<<4,  /* __volatile__ */
  ^~~~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:52:5: note: previous definition of 'IFF_POINTOPOINT' was here
     IFF_POINTOPOINT = 0x10, /* Interface is point-to-point link.  */
     ^~~~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:55:25: error: redeclaration of enumerator 'IFF_NOTRAILERS'
 # define IFF_NOTRAILERS IFF_NOTRAILERS
                         ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:76:2: note: in expansion of macro 'IFF_NOTRAILERS'
  IFF_NOTRAILERS   = 1<<5,  /* sysfs */
  ^~~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:54:5: note: previous definition of 'IFF_NOTRAILERS' was here
     IFF_NOTRAILERS = 0x20, /* Avoid use of trailers.  */
     ^~~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:57:22: error: redeclaration of enumerator 'IFF_RUNNING'
 # define IFF_RUNNING IFF_RUNNING
                      ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:77:2: note: in expansion of macro 'IFF_RUNNING'
  IFF_RUNNING   = 1<<6,  /* __volatile__ */
  ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:56:5: note: previous definition of 'IFF_RUNNING' was here
     IFF_RUNNING = 0x40,  /* Resources allocated.  */
     ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:59:20: error: redeclaration of enumerator 'IFF_NOARP'
 # define IFF_NOARP IFF_NOARP
                    ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:78:2: note: in expansion of macro 'IFF_NOARP'
  IFF_NOARP   = 1<<7,  /* sysfs */
  ^~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:58:5: note: previous definition of 'IFF_NOARP' was here
     IFF_NOARP = 0x80,  /* No address resolution protocol.  */
     ^~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:61:22: error: redeclaration of enumerator 'IFF_PROMISC'
 # define IFF_PROMISC IFF_PROMISC
                      ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:79:2: note: in expansion of macro 'IFF_PROMISC'
  IFF_PROMISC   = 1<<8,  /* sysfs */
  ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:60:5: note: previous definition of 'IFF_PROMISC' was here
     IFF_PROMISC = 0x100, /* Receive all packets.  */
     ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:65:23: error: redeclaration of enumerator 'IFF_ALLMULTI'
 # define IFF_ALLMULTI IFF_ALLMULTI
                       ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:80:2: note: in expansion of macro 'IFF_ALLMULTI'
  IFF_ALLMULTI   = 1<<9,  /* sysfs */
  ^~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:64:5: note: previous definition of 'IFF_ALLMULTI' was here
     IFF_ALLMULTI = 0x200, /* Receive all multicast packets.  */
     ^~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:68:21: error: redeclaration of enumerator 'IFF_MASTER'
 # define IFF_MASTER IFF_MASTER
                     ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:81:2: note: in expansion of macro 'IFF_MASTER'
  IFF_MASTER   = 1<<10, /* __volatile__ */
  ^~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:67:5: note: previous definition of 'IFF_MASTER' was here
     IFF_MASTER = 0x400,  /* Master of a load balancer.  */
     ^~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:70:20: error: redeclaration of enumerator 'IFF_SLAVE'
 # define IFF_SLAVE IFF_SLAVE
                    ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:82:2: note: in expansion of macro 'IFF_SLAVE'
  IFF_SLAVE   = 1<<11, /* __volatile__ */
  ^~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:69:5: note: previous definition of 'IFF_SLAVE' was here
     IFF_SLAVE = 0x800,  /* Slave of a load balancer.  */
     ^~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:73:24: error: redeclaration of enumerator 'IFF_MULTICAST'
 # define IFF_MULTICAST IFF_MULTICAST
                        ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:83:2: note: in expansion of macro 'IFF_MULTICAST'
  IFF_MULTICAST   = 1<<12, /* sysfs */
  ^~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:72:5: note: previous definition of 'IFF_MULTICAST' was here
     IFF_MULTICAST = 0x1000, /* Supports multicast.  */
     ^~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:76:22: error: redeclaration of enumerator 'IFF_PORTSEL'
 # define IFF_PORTSEL IFF_PORTSEL
                      ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:84:2: note: in expansion of macro 'IFF_PORTSEL'
  IFF_PORTSEL   = 1<<13, /* sysfs */
  ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:75:5: note: previous definition of 'IFF_PORTSEL' was here
     IFF_PORTSEL = 0x2000, /* Can set media type.  */
     ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:78:24: error: redeclaration of enumerator 'IFF_AUTOMEDIA'
 # define IFF_AUTOMEDIA IFF_AUTOMEDIA
                        ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:85:2: note: in expansion of macro 'IFF_AUTOMEDIA'
  IFF_AUTOMEDIA   = 1<<14, /* sysfs */
  ^~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:77:5: note: previous definition of 'IFF_AUTOMEDIA' was here
     IFF_AUTOMEDIA = 0x4000, /* Auto media select active.  */
     ^~~~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:80:22: error: redeclaration of enumerator 'IFF_DYNAMIC'
 # define IFF_DYNAMIC IFF_DYNAMIC
                      ^
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:86:2: note: in expansion of macro 'IFF_DYNAMIC'
  IFF_DYNAMIC   = 1<<15, /* sysfs */
  ^~~~~~~~~~~
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:79:5: note: previous definition of 'IFF_DYNAMIC' was here
     IFF_DYNAMIC = 0x8000 /* Dialup device with changing addresses.  */
     ^~~~~~~~~~~
In file included from src/if_monitor.c:30:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:169:8: error: redefinition of 'struct ifmap'
 struct ifmap {
        ^~~~~
In file included from src/if_monitor.c:28:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:111:8: note: originally defined here
 struct ifmap
        ^~~~~
In file included from src/if_monitor.c:30:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:203:8: error: redefinition of 'struct ifreq'
 struct ifreq {
        ^~~~~
In file included from src/if_monitor.c:28:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:126:8: note: originally defined here
 struct ifreq
        ^~~~~
In file included from src/if_monitor.c:30:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/linux/if.h:252:8: error: redefinition of 'struct ifconf'
 struct ifconf  {
        ^~~~~~
In file included from src/if_monitor.c:28:0:
/home/connor/.nerves/artifacts/farmbot_system_rpi3-portable-1.7.2-farmbot.1/staging/usr/include/net/if.h:176:8: note: originally defined here
 struct ifconf
        ^~~~~~
make: *** [Makefile:61: /home/connor/farmbot/os/farmbot_os/farmbot_os/_build/rpi3/rpi3_dev/lib/vintage_net/ebin/../obj/if_monitor.o] Error 1
could not compile dependency :vintage_net, "mix compile" failed. You can recompile this dependency with "mix deps.compile vintage_net", update it with "mix deps.update vintage_net" or clean it with "mix deps.clean vintage_net"

```

--- a/Makefile
+++ b/Makefile
@@ -51,7 +51,16 @@ CC ?= $(CROSSCOMPILE)-gcc
 # Enable for debug messages
 # CFLAGS += -DDEBUG
 
+# Unfortunately, depending on the system we're on, we need
+# to specify -std=c99 or -std=gnu99. The later is more correct,
+# but it fails to build on many setups.
+# NOTE: Need to call sh here since file permissions are not preserved
+#       in hex packages.
+ifeq ($(shell CC=$(CC) sh src/test-c99.sh),yes)
+CFLAGS += -std=c99 -D_XOPEN_SOURCE=600
+else
 CFLAGS += -std=gnu99
+endif
 
 all: install
 

--- a/src/test-c99.sh
+++ b/src/test-c99.sh
@@ -1,0 +1,35 @@
+#!/bin/sh
+
+if [ -z $CC ]; then
+  CC=cc
+fi
+
+# See Makefile
+$CC $CFLAGS -std=c99 -D_XOPEN_SOURCE=600 -o /dev/null -xc - 2>/dev/null <<EOF
+// For whatever reason, this test is more robust when the include
+// order matches what's in netif.c
+#include <err.h>
+#include <poll.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include <arpa/inet.h>
+#include <libmnl/libmnl.h>
+#include <net/if_arp.h>
+#include <net/if.h>
+#include <net/route.h>
+#include <linux/if.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+
+int main(int argc,char *argv[]) {
+    return IFF_UP;
+}
+EOF
+if [ "$?" = "0" ]; then
+    printf "yes"
+fi
+

